### PR TITLE
feat: convert string identifiers in `where` conditions

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

For search for `Documents` to work we need to submit identifiers in `where` condition as `Buffers`

## What was done?

- Modified `douments.get` method
- Implemented tests

## How Has This Been Tested?

Unit tests.

## Breaking Changes

None.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 1]
```diff
@@ -9,12 +11,18 @@ import {Platform} from "../../Platform";
  */
 declare interface fetchOpts {
     where: any;
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:13]: `where?: WhereCondition`

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 2]
```diff
@@ -9,12 +11,18 @@ import {Platform} from "../../Platform";
...
-    orderBy: any;
-    limit: number;
-    startAt: number;
-    startAfter: number;
+    orderBy?: any;
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:14]: Define structure too?

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 3]
```diff
@@ -9,12 +11,18 @@ import {Platform} from "../../Platform";
...
+    limit?: number;
+    startAt?: number;
+    startAfter?: number;
 }
 
+type WhereCondition = [
+    string,
+    string,
+    WhereCondition,
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:23]: `any|WhereCondition` ?

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 4]
```diff
@@ -9,12 +11,18 @@ import {Platform} from "../../Platform";
...
-    startAfter: number;
+    orderBy?: any;
+    limit?: number;
+    startAt?: number;
+    startAfter?: number;
 }
 
+type WhereCondition = [
+    string,
+    string,
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:22]: `'<' | '<=' | '==' | '>' | '>=' | 'in' | 'startsWith' | 'elementMatch' | 'length' | 'contains'`

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 5]
```diff
@@ -31,6 +39,45 @@ const ensureAppContractFetched = async function (this: Platform, appName) {
...
+ *
+ * @param {WhereCondition} whereCondition
+ * @param {Object} binaryProperties
+ * @param {null|string} [parentProperty=null]
+ *
+ * @return {Array}
+ */
+function convertIdentifierProperties(whereCondition: WhereCondition, binaryProperties: Record<string, any>, parentProperty: null|string = null) {
+    const [propertyName, operator, propertyValue] = whereCondition;
+
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:55]: const fullPropertyName = parentProperty ? `${parentProperty}.${propertyName}`: propertyName;

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 6]
```diff
@@ -31,6 +39,45 @@ const ensureAppContractFetched = async function (this: Platform, appName) {
...
+    if (operator === 'elementMatch') {
+        return [
+            propertyName,
+            operator,
+            convertIdentifierProperties(
+                propertyValue,
+                binaryProperties,
+                parentProperty ? `${parentProperty}.${propertyName}`: propertyName,
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:63]: fullPropertyName

File: src/SDK/Client/Platform/methods/documents/get.ts [hunk 7]
```diff
@@ -31,6 +39,45 @@ const ensureAppContractFetched = async function (this: Platform, appName) {
...
+            ),
+        ];
+    }
+
+    const property = binaryProperties[
+        parentProperty ? `${parentProperty}.${propertyName}`: propertyName
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/get.ts:69]: fullPropertyName

**Review by jawid-h — APPROVED** [review]

**Pull request merged**